subsys/bluetooth/controller/ll_sw/ull_llcp_conn_upd.c

```diff
@@ -821,6 +845,44 @@ static void rp_cu_st_idle(struct ll_conn *conn, struct proc_ctx *ctx, uint8_t ev
 }
 
 #if defined(CONFIG_BT_CTLR_CONN_PARAM_REQ)
+static void rp_cu_st_wait_conn_param_req_available(struct ll_conn *conn, struct proc_ctx *ctx,
+						   uint8_t evt, void *param)
+{
+	/* Check if CPR is already active on other connection.
+	 * If so check if possible to send reject right away
+	 * otherwise stay in wait state in case CPR becomes
+	 * available before we can send send reject
+	 */
+	switch (evt) {
+	case RP_CU_EVT_CONN_PARAM_REQ:
+	case RP_CU_EVT_RUN:
+		if (cpr_active_is_set(conn)) {
+			if (!llcp_rr_ispaused(conn) && llcp_tx_alloc_peek(conn, ctx)) {
+				/* We're good to reject immediately */
+				ctx->data.cu.rejected_opcode = PDU_DATA_LLCTRL_TYPE_CONN_PARAM_REQ;
+				ctx->data.cu.error = BT_HCI_ERR_UNSUPP_LL_PARAM_VAL;
+				rp_cu_send_reject_ext_ind(conn, ctx, evt, param);
+			}
+			ctx->state = RP_CU_STATE_WAIT_CONN_PARAM_REQ_AVAILABLE;
```

Comment:
This will override the `ctx->state` set by  `rp_cu_complete` called from `rp_cu_send_reject_ext_ind`. I know the procedure is marked as complete, and it probably doesn't do anything, but still for consistency i think the sake should be set to the intended  `RP_CU_STATE_IDLE` to avoid any side-effects by an unexpected `RP_CU_EVT_RUN`  event.